xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-rss/src/main/java/org/xwiki/rendering/internal/macro/rss/RssMacro.java

```diff
@@ -152,6 +149,16 @@ public boolean supportsInlineMode()
         return result;
     }
 
+    @Override
+    protected List<Block> parseContent(RssMacroParameters parameters, String content,
+        MacroTransformationContext context) throws MacroExecutionException
+    {
+        if (content == null) {
+            content = StringUtils.EMPTY;
```

Comment:
I doubt checkstyle like this much. Also I'm not sure what you feel the need to do this since parse() starts with a StringUtils.isEmpty check.

Reply:
about checkstyle it was ok when running it with a `Pquality` so I guess it's ok. I did this because it was checked and changed in the original execute() method. Now I need it since I don't manage the same way null and empty content in execute.

Reply:
> I don't manage the same way null and empty content in execute

I don't understand what you mean. This code is making any null 'content` an empty String so you definitely do the same thing for empty string and null. Also my point is that this check is not needed because parse already deal with that.

Reply:
Was talking about `AbstractBoxMacro#execute`. Here I just kept the check that was already done in `execute(parameters, content == null ? StringUtils.EMPTY : content, context);` as I didn't want to take any risk for a bugfix.

Reply:
Actually parseContent will never been called with null content because:
* you already check for null content in execute
* AbstractBoxMacro never call parseContent with null content

So I really don't understand why you need this if.